pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -609,6 +575,48 @@ func (c *constraint) listOrderedSCCs(
 	return constraints, nil
 }
 
+func (c *constraint) waitForReadyState(ctx context.Context) error {
+	const (
+		interval  = 1 * time.Second
+		timeout   = 10 * time.Second
+		immediate = true
+	)
+
+	err := wait.PollUntilContextTimeout(ctx, interval, timeout, immediate, c.areListersSynced)
+	if err != nil {
+		return fmt.Errorf("securitycontextconstraints.security.openshift.io cache is not synchronized")
+	}
+
+	// wait a few seconds until the synchronized list returns all the required SCCs created by the kas-o.
+	// If this doesn't happen, then indicate which ones are missing.  This seems odd, but our CI system suggests that this happens occasionally.
+	// If the SCCs were all deleted, then no pod will pass SCC admission until the SCCs are recreated, but the kas-o (which recreates them)
+	// bypasses SCC admission, so this does not create a cycle.
+	var requiredSCCErr error
+	err = wait.PollUntilContextTimeout(ctx, interval, timeout, immediate, func(context.Context) (bool, error) {
+		if requiredSCCErr = requireStandardSCCs(c.sccLister.List(labels.Everything())); requiredSCCErr != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		if requiredSCCErr != nil {
+			return requiredSCCErr
+		}
+		return fmt.Errorf("securitycontextconstraints.security.openshift.io required check failed oddly")
+	}
+
+	return nil
+}
+
+func (c *constraint) areListersSynced(_ context.Context) (bool, error) {
```

Comment:
That's a nice refactoring; I would even inline this to an anonymous func on the `PollUntilContextTimeout()` since it's simple enough and not used elsewhere. This will make clear also why that func returns a bool and an error but the error is always nil.

Just a minor suggestion, happy with this too.